cmake/modules/RootMacros.cmake

```diff
@@ -282,6 +282,95 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
     set(CMAKE_INSTALL_LIBDIR ${CMAKE_CURRENT_BINARY_DIR})
     set(libprefix "")
   endif()
+  
+  
+  if((CMAKE_PROJECT_NAME STREQUAL ROOT) AND (TARGET ${ARG_MODULE}))
+    set(incdirs)
+
+    # for a time been, roocling makes strange error
+    list(APPEND incdirs ${CMAKE_BINARY_DIR}/include)
+    
+    if(TARGET ${ARG_MODULE})
+       get_target_property(target_incdirs ${ARG_MODULE} INCLUDE_DIRECTORIES)
+       foreach(dir ${target_incdirs})
+           list(APPEND incdirs ${dir})
+       endforeach()
+    endif()
+
+    foreach(dir ${ARG_INCLUDES})
+        list(APPEND incdirs ${dir})
+    endforeach()
+
+    list(REMOVE_DUPLICATES incdirs)
+    
+    foreach(d ${incdirs})
+       list(APPEND includedirs -I${d})
+    endforeach()
+    
+    set(pureincdirs ${incdirs})
+    
+    set(headerfiles)
+    set(_list_of_header_dependencies)
+    foreach(fp ${ARG_UNPARSED_ARGUMENTS})
+      if(${fp} MATCHES "[*?]") # Is this header a globbing expression?
+        file(GLOB files inc/${fp} ${fp}) # Elements of ${fp} have the complete path.
```

Comment:
Where does all this code come from? Why was it not needed before?

Reply:
I just copy and paste old code. May be there is more elegant way - but it works. One always can improve it later

Reply:
I would like to avoid this code duplication.